Packs/BitwardenPasswordManager/Integrations/BitwardenPasswordManager/BitwardenPasswordManager.py

```diff
@@ -0,0 +1,236 @@
+            data=json_data
+        )
+
+        new_access_token = access_token_obj.get('access_token', '')
+        expire_in = arg_to_number(access_token_obj.get('expires_in')) or 1
+        self.store_token_in_context(new_access_token, expire_in)
+
+        return new_access_token
+
+    def store_token_in_context(self, token: str, expire_in: int) -> None:
+        expire_date = get_current_time() + timedelta(seconds=expire_in) - timedelta(seconds=SECONDS_BEFORE_TOKEN_EXPIRED)
+        set_integration_context(context={
+            'token': token,
+            'expires': str(expire_date)
+        })
+
+    def get_events(self, start_date: str = '', continuation_token: str = '') -> dict:
+        end_date_datetime_object = get_current_time() + timedelta(days=1)
+        end_date = end_date_datetime_object.strftime(
+            '%Y-%m-%dT%H:%M:%S.') + f'{end_date_datetime_object.microsecond // 10000:02d}'
+
+        params = {
+            'start': start_date,
+            'end': end_date
+        }
+
+        if continuation_token:
+            params['continuationToken'] = continuation_token
+
+        headers = {'Authorization': f'Bearer {self.token}'}
+
+        res = self._http_request(
+            method='GET',
+            url_suffix='/public/events',
+            headers=headers,
+            params=params
+        )
+
+        return res
+
+
+def test_module(client: Client) -> str:
+    event, _ = fetch_events(client, max_fetch=1)
+    if not event:
+        raise ValueError('failed to fetch events')
+
+    return 'ok'
+
+
+def get_events_command(client: Client, start_date_arg: str, max_fetch: int) -> tuple:
+    events, _ = fetch_events(client=client, max_fetch=max_fetch, start_date_str=start_date_arg)
+    if events:
+        events = events[:max_fetch]
+        return events, CommandResults(readable_output=tableToMarkdown('Bitwarden Events', events),
+                                      raw_response=events)
+
+    return [], CommandResults(readable_output='No events found')
+
+
+def fetch_events(client: Client, max_fetch: int, start_date_str: str = DEFAULT_FIRST_FETCH) -> tuple[
+    List[Dict[str, Any]], Dict[str, Any]]:
+    last_run = demisto.getLastRun()
+    events, continuation_token = get_events(client, max_fetch, start_date_str, last_run)
+    unique_events = get_unique_events(events, last_run)
+    latest_events = filter_events(events=events, latest=True)
+    hash_latest_events = hash_events(latest_events)
+    if continuation_token:
+        demisto.debug(
+            f'Bitwarden - Fetched {len(unique_events)} which is the maximum or greater then the number of events.'
+            f' Will keep the fetching in the next fetch.')
+        created = start_date_str or last_run.get('last_fetch') or (
+            (get_current_time() - timedelta(minutes=1)).strftime(DATE_FORMAT))
+        new_last_run = {'continuationToken': continuation_token, 'last_fetch': created, 'nextTrigger': '0',
```

Comment:
just to make sure: we don't need the same end_time for the next run when using the continuationToken?

Reply:
I think it is important, but how can the end time change?
For the fetch command it is not possible to add an end time and for the get command the end time is fixed.
Since I set end time to be the default end time (or what the user enter) it will be constant.